lib/deref.js

```diff
@@ -179,7 +179,11 @@ module.exports = {
       // This nonsense is needed because the schemaFaker doesn't respect options.maxItems/minItems
       schema.maxItems = 2;
       schema.minItems = 2;
-      schema.items = this.resolveRefs(schema.items, bodyTypeOption, components, stack);
+      // have to create a shallow clone of schema object
+      // so that the original schema.items object will not change which was leading to circular references
```

Comment:
@Dhroov7 No - schema.items changing was causing errors while dealing with circular references. It was not leading to circular references.